resource/config.go

```diff
@@ -26,6 +26,7 @@ type Config struct {
 
 	ConvertedAttributes ConfigValidator
 	ImplicitDependsOn   []string
+	ProvidedBy          string
```

Comment:
This is not used yet. Not sure how I feel about it. This would store the name of the module that provides this and would be part of the `Dependencies()`